### ISSUE TITLE
Detect backend switching and warn

### ISSUE THREAD

**Issue by AbelLykens** [lightningnetwork/lnd#6901]
### Background

Switching backends is [not supported](https://github.com/lightningnetwork/lnd/issues/2526#issuecomment-456635169):
> This isn't supported. You can't switch to neutrino as no headers are on disk. As is you can only switch between full node operating nodes, and not also neutrino.

It's still easily possible to do this (configure a different backend). A health check detecting an unsupported change would be good.

### Expected behaviour

1. Change backend
2. Get a warning: this is not supported

### Actual behaviour

1. Change backend
2. Problems! #6900  --  #2526 --  #6572 -- more!

**Comment by positiveblue** [comment]
Solved in [6905](https://github.com/lightningnetwork/lnd/pull/6905), thanks @AbelLykens

**Comment by AbelLykens** [comment]
@positiveblue Yeah partially. The docs are a start (warn in text), but having a health check could still be useful. Please consider.

If you/we believe this doc change is enough that's fine too ;)
